.github/workflows/build-and-test-device.yaml

```diff
@@ -156,4 +157,4 @@ jobs:
         if: matrix.config.label == 'with-cuda'
         run: |
           cd python
-          python3 -m pytest -vv
+          python3 -m pytest -vv tests
```

Comment:
Unfortunately pytest does discover tests recursively from the subproject if you try discovery from the root of the project; need to see if there's a more robust way of handling this

Reply:
`norecursedirs` in `pytest.ini`, like this: https://github.com/scipy/scipy/blob/main/pytest.ini#L3 ?